### DISCUSSION TITLE
dscmd xlsx - set column data type the same as Power BI dataset

### DISCUSSION THREAD

**Discussion by tnhurst57** [DaxStudio/DaxStudio#1319] [Q&A]
Is there a way, when exporting to an xlsx file with dscmd, to set the resulting column to be the same as the Power BI dataset column?

**Comment by dgosbell** [comment]
Excel does not really have a concept of column data types. Are you possibly talking about the formatting? It looks like there might be a regression in the way formatting is applied

**Comment by tnhurst57** [comment]
Hey Darren,
I am a big fan of dscmd. Here is the issue I am trying to solve:
I have a Power BI app that has a Power Automate button. This button executes a flow that export key fields into a file.
Another process reads that file (serial numbers in this case), and uses these to query the Power BI dataset(s) using dscmd to a CSV or XLSX file.
Finally, a Python script reads that XLSX file into a Pandas dataframe. I am trying to ultimately get the columns in the dataframe to be the same data type as the original Power BI dataset. I have found errors in the Power Automate export - which led me to dscmd.

Any recommendations?

**Comment by dgosbell** [comment]
The problem here is that neither the csv or xslx file formats are strongly typed. If you could use a Fabric notebook you could query the semantic model on the service and load the results directly into a dataframe. Otherwise, you are going to have to explicitly set the types yourself.
